### PR TITLE
(DOCSP-39526): Consolidate Authenticate Users page

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info - SDK Docs Consolidation

Jira ticket: https://jira.mongodb.org/browse/DOCSP-39526

*Staged Page*

- [Authenticate Users](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-39526/sdk/users/authenticate-users/)
- [Manage User Tokens](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-39526/sdk/users/manage-user-tokens/): New page that moves the user access token info off the "Authenticate" page. This is currently [a separate page in Flutter](https://www.mongodb.com/docs/atlas/device-sdks/sdk/flutter/users/access-token/). As devs shouldn't have to handle this manually for most use cases, _and_ the OTP is getting super long, I thought it made sense to move the info off to a separate page.

*Page Source*

Add links to every SDK's pages where you got the SDK-specific information:

- [C++: Authenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/cpp/users/authenticate-users/)
- [Flutter: Authenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/flutter/users/authenticate/)
- [Flutter: Get an Access Token](https://www.mongodb.com/docs/atlas/device-sdks/sdk/flutter/users/access-token/)
- [Java: Authenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/java/users/authenticate-users/)
- [Kotlin: Autenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/kotlin/users/authenticate-users/)
- [.NET: Authenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/dotnet/manage-users/authenticate/)
- [Node.js: Authenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/node/users/authenticate-users/)
- [Swift: Authenticate Users](https://www.mongodb.com/docs/atlas/device-sdks/sdk/swift/users/authenticate-users/)

### PR Author Checklist

Before requesting a review for your PR, please check these items:

- [x] Open the PR against the `feature-consolidated-sdk-docs` branch instead of `master`
- [x] Tag the consolidated page for:
  - genre
  - meta.keywords
  - meta.description

#### Naming
- [x] Update Realm naming and the language around persistence layer/local/device per [this document](https://docs.google.com/document/d/126OczVxBWAwZ4P5ZsSM29WI3REvONEr1ald-mAwPtyQ/edit?usp=sharing)
- [x] Include `.rst` files comply with [the naming guidelines](https://docs.google.com/document/d/1h8cr66zoEVeXytVfvDxlCSsUS5IZwvUQvfSCEXNMpek/edit#heading=h.ulh8b5f2hu9)

#### Links and Refs
- [x] Create new consolidated SDK ref targets starting with "_sdks-" for relevant sections
- [x] Remove or update any SDK-specific refs to use the new consolidated SDK ref targets
- [x] [Update any Kotlin API links](https://jira.mongodb.org/browse/DOCSP-32519) to use the new Kotlin SDK roles

#### Content
- [x] Shared code boxes have snippets or placeholders for all 9 languages
- [x] API description sections have API details or a generic placeholder for all 9 languages
- [x] Check related pages for relevant content to include
- [x] Create a ticket for missing examples in each relevant SDK: Consolidation Gaps epic

### Reviewer Checklist

As a reviewer, please check these items:

- [ ] Shared code example boxes contain language-specific snippets or placeholders for every language
- [ ] API reference details contain working API reference links or generic content
- [ ] Realm naming/language has been updated
- [ ] All relevant content from individual SDK pages is present on the consolidated page

### CODE REVIEW

File: source/sdk/users/authenticate-users.txt [hunk 1]
```diff
@@ -4,20 +4,880 @@
...
+
+   .. tab::
+      :tabid: cpp-sdk
+
+      .. include:: /includes/api-details/cpp/users/authenticate-get-current-user-description.rst
+
+   .. tab::
+      :tabid: csharp
+
+      .. include:: /includes/api-details/csharp/users/authenticate-log-in-user-description.rst
```
Comment (krollins-mdb) [inline source/sdk/users/authenticate-users.txt:856]: I assume you wanted this include instead?
```suggestion
      .. include:: /includes/api-details/csharp/users/authenticate-get-current-user-description.rst
```

File: source/sdk/users/authenticate-users.txt [hunk 2]
```diff
@@ -4,20 +4,880 @@
...
+      
+   .. tab::
+      :tabid: dart
+
+      .. include:: /includes/api-details/dart/users/authenticate-log-in-user-description.rst
```
Comment (krollins-mdb) [inline source/sdk/users/authenticate-users.txt:861]: I assume you wanted this include instead?
```suggestion
      .. include:: /includes/api-details/dart/users/authenticate-get-current-user-description.rst
```

File: source/includes/api-details/swift/users/authenticate-register-create-new-user-account-description.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
+When a user registers or successfully authenticates to your App, Atlas App
+Services automatically creates a :swift-sdk:`User
+Extensions/User.html>` object, which contains a unique
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/includes/api-details/swift/users/authenticate-register-create-new-user-account-description.rst]: Oops, yes! Good catch - thanks!

File: source/sdk/users/authenticate-users.txt [hunk 4]
```diff
@@ -4,20 +4,880 @@
...
 -------------
 
-.. _sdks-refresh-token-expiry:
+To authenticate users through Atlas App Services, you must have an 
+App Services App with one or more authentication providers enabled.
+
+To set up an App Services App with authentication providers, 
+complete the following:
+
+#. :ref:`Connect your App to Atlas App Services <sdks-connect-to-atlas>`
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/users/authenticate-users.txt]: Hmm, no, you're right. I didn't audit this page for app/App usage when I brought the content over and it seems there may be some misses here. I'll go back and give this another pass.

File: source/sdk/users/authenticate-users.txt [hunk 5]
```diff
@@ -4,20 +4,880 @@
...
+#. :ref:`Enable and Configure Authentication Providers 
+   <authentication-providers>` in the App Services documentation
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/users/authenticate-users.txt:72]: Ahh, gotcha. Hah, I know Cory feels pretty strongly about keeping the "in the App Services documentation" so I'll wordsmith ;)

File: source/sdk/users/authenticate-users.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/users/authenticate-users.txt]: Oops, yes, thank you! 🙇

**Pull request merged**